webpack.config.ts

```diff
@@ -0,0 +1,234 @@
+ * supported. The order of this array is important, as the order of outputted styles will match. Svelte component
+ * styles will always appear last in the bundle.
+ */
+let stylesheets = ['./src/styles/index.scss'];
+
+const config: webpack.Configuration & WebpackDevServer.Configuration = {
+	entry: {
+		bundle: [
+			// Note: Paths in the `stylesheets` variable will be added here automatically
+			'./src/main.ts',
+		],
+	},
+	resolve: {
+		alias: {
+			// Note: Additional aliases will be loaded automatically from `tsconfig.compilerOptions.paths`
+			svelte: path.resolve('node_modules', 'svelte'),
+		},
+		extensions: ['.mjs', '.js', '.ts', '.svelte'],
+		mainFields: ['svelte', 'browser', 'module', 'main'],
+	},
+	output: {
+		publicPath: '/build/',
+		path: __dirname + '/public/build',
+		filename: '[name].js',
+		chunkFilename: '[name].[id].js',
+	},
+	module: {
+		rules: [
+			{
+				test: /\.svelte$/,
+				use: {
+					loader: 'svelte-loader-hot',
+					options: {
+						dev: !prod,
+						emitCss: prod,
+						hotReload: !prod,
+						hotOptions: {
+							// List of options and defaults: https://www.npmjs.com/package/svelte-loader-hot#usage
+							noPreserveState: false,
+							optimistic: true,
+						},
+						preprocess: Preprocess({
+							scss: true,
+							postcss: {
+								plugins: [require('autoprefixer')],
+							},
+						}),
+					},
+				},
+			},
+			{
+				test: /\.(scss|sass)$/,
+				use: [
+					{
+						loader: MiniCssExtractPlugin.loader,
+						options: {
+							hmr: !prod,
+							sourceMap: !prod || sourceMapsInProduction,
+						},
+					},
+					'css-loader',
+					{
+						loader: 'postcss-loader',
+						options: {
+							plugins: [require('autoprefixer')],
+						},
+					},
+					'sass-loader',
+				],
+			},
+			{
+				test: /\.css$/,
+				use: [
+					{
+						loader: MiniCssExtractPlugin.loader,
+						options: {
+							hmr: !prod,
+							sourceMap: !prod || sourceMapsInProduction,
+						},
+					},
+					'css-loader',
+				],
+			},
+			{
+				test: /\.ts$/,
+				use: 'ts-loader',
+				exclude: /node_modules/,
+			},
+		],
+	},
+	devServer: {
+		hot: true,
+		stats: 'minimal',
+		contentBase: 'public',
+		watchContentBase: true,
+	},
+	mode,
+	plugins: [
+		new MiniCssExtractPlugin({
+			filename: '[name].css',
+		}),
+	],
+	optimization: {
+		minimizer: [],
+	},
+	devtool: prod && !sourceMapsInProduction ? false : 'source-map',
+};
+
+// Add stylesheets to the build
+if (Array.isArray(stylesheets) || typeof stylesheets === 'string') {
+	if (!Array.isArray(stylesheets)) {
+		stylesheets = [stylesheets];
+	}
+
+	// Explicitly type bundle as string[]
+	type Replace<T, K> = Omit<T, keyof K> & K;
+	config.entry = config.entry as Replace<
+		typeof config.entry,
+		{
+			bundle: string[];
+		}
+	>;
+	(config.entry.bundle as string[]).unshift.apply(
+		config.entry.bundle,
+		stylesheets
+	);
```

Comment:
I am not happy about this typecast but it does the job... again, feel free to reject this PR, no hard feelings.